### ISSUE TITLE
Request with correct API key does not trigger controller action

### ISSUE THREAD

**Issue by jeroenwo** [jamesharling/AspNet.Security.ApiKey.Providers#2]
When implementing the ApiKey middleware, the request is checked for an API key. The unit tests verify that, but the actual controller action is never triggered.

Added a unit test which fails:
```csharp
[TestMethod]
public async Task Access_Protected_Resource_Using_Valid_ApiKey_Authentication_Should_Return_Result()
{
    var response = await this.api.CreateRequest("/api/authenticated/values").AddHeader("Authorization", "ApiKey 123").GetAsync();
    var data = await response.Content.ReadAsStringAsync();

    data.Should().NotBeNullOrWhiteSpace();
}
```
It seems like the result is returned immediately after checking the API key?

**Comment by jeroenwo** [comment]
If I use the following code in the `ApiKeyHandler`, it works:
```csharp
if (validateApiKeyContext.Result != null)
{
    this.Logger.ApiKeyValidationSucceeded();

    var claims = new[] { new Claim(ClaimTypes.Name, "aa") };
    var identity = new ClaimsIdentity(claims, Scheme.Name);
    var principal = new ClaimsPrincipal(identity);
    var ticket = new AuthenticationTicket(principal, Scheme.Name);

    return AuthenticateResult.Success(ticket);
}
```
However, if I reuse the created `ClaimsPrincipal`, it doesn't:
```csharp
if (validateApiKeyContext.Result?.Principal != null)
{
    this.Logger.ApiKeyValidationSucceeded();

    var ticket = new AuthenticationTicket(validateApiKeyContext.Result.Principal, Scheme.Name);

    return AuthenticateResult.Success(ticket);
}
```
Neither does reusing the created `AuthenticationTicket`:
```csharp
if (validateApiKeyContext.Result?.Ticket != null)
{
    this.Logger.ApiKeyValidationSucceeded();
                    
    return AuthenticateResult.Success(validateApiKeyContext.Result.Ticket);
}
```
Any ideas?

**Comment by wind-raven** [comment]
in my authorize attribute I had to add AuthenticationSchemes = ApiKeyDefaults.AuthenticationScheme to get the controller actions to trigger with out any changes.

**Comment by farlee2121** [comment]
My experimenting suggests the issues is the ClaimsIdentity. OnApiKeyValidated makes you define a principle, but it does not make you create a ClaimsIdentity. If the authentication scheme on the identity does not match the apikey scheme, then the request will return 403 Forbidden.

In short, the solution is to set
`context.Principal = new System.Security.Claims.ClaimsPrincipal(new System.Security.Claims.ClaimsIdentity(context.Scheme.Name));`

in OnApiKeyValidated

**Comment by jamesharling** [comment]
Looks like we need a small tweak to the Handler and the `ApiKeyValidated` event. Will look to get this done soon.

**Comment by carlosmorcerf** [comment]
any update?

**Comment by jamesharling** [comment]
Now fixed in v1.1.0. Should be available on NuGet once it has been validated by them.
